### DISCUSSION TITLE
How to click a button and open it in a new tab

### DISCUSSION THREAD

**Discussion by elcobenoz** [go-rod/rod#310] [General]
Hi,
I got an element that can be clicked but I wish to open that in a new tab, how can i do this?

I have seen that u can use ```.MustPage()``` but this opens a new tab, I want the button when clicked to open a new tab (like right clicking, open in new tab normally).

```page2.MustElement(".ProductCard__EnterDrawButtonSC-sc-1dh591-6.IKfNn").MustClick()```

**Comment by ysmood** [comment]
```go
page.MustElement("a").MustEval(`this.target = "_blank"`).MustClick()
```

**Reply by elcobenoz** [reply]
I get this error:

2-valued page2.MustElement(".ProductCard__EnterDrawButtonSC-sc-1dh591-6.IKfNn").Eval(`this.target = "_blank"`) (value of type (*proto.RuntimeRemoteObject, error)) where single value is expectedcompiler

I do not understand why.
I forgot to mention that page2 is *rod.Element (made from the example in the docs under "select lists of elements" , basically it is  the same as the variable "project" in that documented example)

**Reply by ysmood** [reply]
```
page.MustElement("a").MustEval(`this.target = "_blank"`).MustClick()
```

**Reply by ysmood** [reply]
Friend, it's a syntax error, you should be able to fix it by googling.

**Reply by ysmood** [reply]
Try to google why I code like this, use your brain to think, not just copy code.

**Reply by elcobenoz** [reply]
> Try to google why I code like this, use your brain to think, not just copy code.

I do try to think, doesnt help when the documentation isnt too clear. Ive tried looking into the godocs as well and it doesnt have much information either on what MustEval does, what i could pickup from that is adding .MustClick() doesnt work cause it is not a gson.JSON type

**Reply by ysmood** [reply]
Sorry, the doc is not completely clear, you can just use the IDE to jump to the source code, that's a convention for Go, the source code and unit test are also part of the doc. I spend a lot of time to make the source code very easy to read and understand.
